src/manifold/src/impl.cpp

```diff
@@ -478,6 +478,37 @@ Manifold::Impl::Impl(const Mesh& mesh, const MeshRelationD& relation,
   Finish();
 }
 
+void Manifold::Impl::Hull(const std::vector<glm::vec3>& vertPos) {
+  size_t numVert = vertPos.size();
+  if (numVert < 4) {
+    status_ = Error::InvalidConstruction;
+    return;
+  }
+
+  Vec<glm::dvec3> pointCloudVec(numVert);
+  manifold::transform(vertPos.begin(), vertPos.end(), pointCloudVec.begin(),
+                      [](const glm::vec3& v) { return glm::dvec3(v); });
+  QuickHull qh(pointCloudVec);
+  ConvexHull hull = qh.getConvexHullAsMesh(pointCloudVec, false);
+  vertPos_.resize(hull.vertices.size());
+  manifold::transform(hull.vertices.begin(), hull.vertices.end(),
+                      vertPos_.begin(),
+                      [](const glm::dvec3& v) { return glm::vec3(v); });
+  halfedge_ = std::move(hull.halfEdges);
+  for_each(autoPolicy(halfedge_.size(), 1e5), countAt(0_uz),
+           countAt(halfedge_.size()),
+           [this](size_t i) { halfedge_[i].face = i / 3; });
+  meshRelation_.originalID = ReserveIDs(1);
+  CalculateBBox();
+  SetPrecision(bBox_.Scale() * kTolerance);
+  SplitPinchedVerts();
+  CalculateNormals();
+  InitializeOriginal();
+  CreateFaces({});
+  SimplifyTopology();
```

Comment:
I agree with your analysis, but let's leave it to the next PR to optimize exactly what's needed here.